### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by zemlyanoi**
## Description
Refactoring of relationship service
Fix performance for getting relationships with candidates

## Tests
- [ ] I have included unit tests 
- [ ] I have included functional tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the google code style of this project.
- [ ] I have ran all tests for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check SonarQube after the build completes, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+      ScreeningRelationship relationship) {
+    String clientId = relationship.getClientId();
+    relationship.setClientId(relationship.getRelativeId());
+    relationship.setRelativeId(clientId);
+    relationship.setRelationshipType(getOppositeSystemCode(relationship.getRelationshipType()));
+    return relationship;
+  }
+
+  private void initSystemCodes() {
+    if (!MapUtils.isEmpty(codesMappedByDescription)) {
```
Comment (OlegKorniichuk) [inline src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java]: just suggestion, once we have MapUtils it has isNotEmpty utility method

File: src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+    if (StringUtils.isEmpty(screeningId)) {
+      return Collections.emptyList();
+    }
+
+    // get participants by screeningId
+    Set<String> legacyClientIds = getLegacyClientIdsByScreeningId(screeningId);
+    // get relationships from legacy
+    Set<ClientRelationship> lagacyRelationships = getCmsRelationships(legacyClientIds);
+    // get relationships from pgsql
+    List<Relationship> nsRelationships = getNsRelationships(screeningId);
```
Comment (OlegKorniichuk) [inline src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java]: call to different data sources can be done in parallel way to improve performance

File: src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+        getRelationshipsThatShouldBeCreated(lagacyRelationships, nsRelationships);
+    result.addAll(createRelationships(shouldBeCreated, screeningId));
+    result.addAll(updateRelationships(shouldBeUpdated));
+    result = getRelationshipsThatShouldNotBeUpdated(result, nsRelationships);
+
+    // select and return
+    if (participantDao.getSessionFactory().getCurrentSession().getTransaction().getStatus()
+        == TransactionStatus.ACTIVE) {
+      participantDao.getSessionFactory().getCurrentSession().flush();
+    }
```
Comment (OlegKorniichuk) [inline src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java]: not sure that it's reasonable to do here but DAOs have grabSession method that we use across the project

File: src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+    relationshipFacadeData.participantsLegacyDescriptors = participantsLegacyDescriptors;
+    relationshipFacadeData.screeningId = screeningId;
+    relationshipFacadeData.screeningParticipants = screeningParticipants;
+    return relationshipFacadeData;
+  }
+
+  private Map<String, ParticipantEntity> getMappedParticipantsById(Set<String> participantIds) {
+    Map<String, ParticipantEntity> map = new HashMap<>();
+    List<ParticipantEntity> participantEntities = participantDao.findByIds(participantIds);
+    if (CollectionUtils.isEmpty(participantEntities)) {
```
Comment (OlegKorniichuk) [inline src/main/java/gov/ca/cwds/rest/services/relationship/RelationshipFacadeLegacyAndNewDB.java]: code is more readable with one approach to test empty/not empty collections. MapUtils vs CollectionUtils
